plugins/modules/user_role_workflow_manager.py

```diff
@@ -1379,27 +1417,40 @@ def get_diff_merged(self, config):
                 self.valid_user_config_parameters(config).check_return_status()
                 (consolidated_data, update_required_param) = self.user_requires_update(self.have["current_user_config"], self.have["current_role_id_config"])
 
-                if not consolidated_data:
-                    self.msg = "User with username '{0}' already exists and does not require an update.".format(self.have.get("username"))
-                    self.no_update_user.append(self.have.get("username"))
-                    self.log(self.msg, "INFO")
-                    responses["role_operation"] = {"response": config}
-                    self.result["response"] = self.msg
-                    self.status = "success"
-                    return self
-
-                if update_required_param.get("role_list"):
-                    if self.want["username"] not in self.have["current_user_config"]["username"]:
-                        task_response = {"error_message": "Username for an existing user cannot be updated."}
+                if self.want.get("password_update"):
+                    if update_required_param.get("role_list"):
+                        if self.want["username"] not in self.have["current_user_config"]["username"]:
+                            task_response = {"error_message": "Username for an existing user cannot be updated."}
+                        else:
+                            self.get_diff_deleted(self.want)
+                            update_required_param["password"] = self.want.get("password")
+                            user_info_params = self.snake_to_camel_case(update_required_param)
+                            task_response = self.create_user(user_info_params)
                     else:
-                        user_in_have = self.have["current_user_config"]
-                        update_param = update_required_param
-                        update_param["user_id"] = user_in_have.get("user_id")
-                        user_info_params = self.snake_to_camel_case(update_param)
-                        task_response = self.update_user(user_info_params)
+                        task_response = {"error_message": "The role name in the user details role_list is not present in the Cisco Catalyst Center,"
+                                         " Please provide a valid role name"}
                 else:
-                    task_response = {"error_message": "The role name in the user details role_list is not present in the Cisco Catalyst Center,"
-                                     " Please provide a valid role name"}
+                    if not consolidated_data:
+                        self.msg = "User with username '{0}' already exists and does not require an update.".format(self.have.get("username"))
```

Comment:
First get username.. Used in other places...
    username = self.have.get("username")

Reply:
Updated the code accordingly.